software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -155,29 +155,42 @@ public void attachLifecycleEffectors(Entity entity) {
                 .build();
     }
 
+    /** @see {@link #newStartEffector()} */
+    public Effector<Void> newSuspendEffector() {
+        return Effectors.effector(Void.class, "suspend")
+                .description("Suspend the process/service represented by an entity")
+                .parameter(StopSoftwareParameters.STOP_PROCESS_MODE)
+                .parameter(StopSoftwareParameters.STOP_MACHINE_MODE)
+                .impl(newSuspendEffectorTask())
+                .build();
+    }
+
     /**
-     * Returns the {@link TaskFactory} which supplies the implementation for the start effector.
+     * Returns the {@link EffectorBody} which supplies the implementation for the start effector.
      * <p>
      * Calls {@link #start(Collection)} in this class.
      */
     public EffectorBody<Void> newStartEffectorTask() {
-        return new EffectorBody<Void>() {
-            @Override
-            public Void call(ConfigBag parameters) {
-                Collection<? extends Location> locations  = null;
+        return new StartEffectorBody();
```

Comment:
I think both re-introducing the effector anonymous classes (in inaccessible methods) or reverting is a valid way to go.
The effect should be quite limited (I'd say non-existent) because users have to explicitly re-configure the key with a `LifecycleEffectorTasks` instance. Using it as a default value doesn't get it serialized.